does not compile on windows

I'm getting this error when trying to use libtcc on windows 10:

    error: failed to run custom build command for `libtcc v0.2.0`

    Caused by:
      process didn't exit successfully: `target\debug\build\libtcc-a6841ebbe154479f\build-script-build` (exit code: 1)
    warning: build failed, waiting for other jobs to finish...
    error: build failed

I figured out that `cargo build -vv` also displays the build.rs output.
Here it is:

    [libtcc 0.2.0] OPT_LEVEL = Some("1")
    [libtcc 0.2.0] TARGET = Some("x86_64-pc-windows-msvc")
    [libtcc 0.2.0] HOST = Some("x86_64-pc-windows-msvc")
    [libtcc 0.2.0] CC_x86_64-pc-windows-msvc = None
    [libtcc 0.2.0] CC_x86_64_pc_windows_msvc = None
    [libtcc 0.2.0] HOST_CC = None
    [libtcc 0.2.0] CC = None
    [libtcc 0.2.0] CFLAGS_x86_64-pc-windows-msvc = None
    [libtcc 0.2.0] CFLAGS_x86_64_pc_windows_msvc = None
    [libtcc 0.2.0] HOST_CFLAGS = None
    [libtcc 0.2.0] CFLAGS = None
    [libtcc 0.2.0] CRATE_CC_NO_DEFAULTS = None
    [libtcc 0.2.0] CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
    [libtcc 0.2.0] DEBUG = Some("true")
    [libtcc 0.2.0] running "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Professional\\VC\\Tools\\MSVC\\14.10.25017\\bin\\HostX64\\x64\\cl.exe" "src/libtcc_test.c" "-o" "C:\\Users\\SARNOW4\\Home\\Code\\Rust\\analyzer\\target\\debug\\build\\libtcc-277853c1fdc58435\\out\\tcc-tmp\\a.out" "-Isrc/tcc-0.9.27" "-ltcc" "-ldl" "-lrt" "-lm" "-lpthread"
    [libtcc 0.2.0] Microsoft (R) C/C++-Optimierungscompiler Version 19.10.25019 f�r x64
    [libtcc 0.2.0] Copyright (C) Microsoft Corporation. Alle Rechte vorbehalten.
    [libtcc 0.2.0]
    [libtcc 0.2.0] cl : Befehlszeile warning D9035 : Die Option "o" ist veraltet und wird in einer der n�chsten Versionen entfernt.
    [libtcc 0.2.0] cl : Befehlszeile warning D9002 : Unbekannte Option "-ltcc" wird ignoriert.
    [libtcc 0.2.0] cl : Befehlszeile warning D9002 : Unbekannte Option "-ldl" wird ignoriert.
    [libtcc 0.2.0] cl : Befehlszeile warning D9002 : Unbekannte Option "-lrt" wird ignoriert.
    [libtcc 0.2.0] cl : Befehlszeile warning D9002 : Unbekannte Option "-lm" wird ignoriert.
    [libtcc 0.2.0] cl : Befehlszeile warning D9002 : Unbekannte Option "-lpthread" wird ignoriert.
    [libtcc 0.2.0] libtcc_test.c
    [libtcc 0.2.0] src/libtcc_test.c(4): fatal error C1083: Datei (Include) kann nicht ge�ffnet werden: "stdlib.h": No such file or directory
    [libtcc 0.2.0] ERROR: Can not find libtcc.a in your host:
    [libtcc 0.2.0]  Tcc should be installed in host when your build target is same as host,
    [libtcc 0.2.0]  because libtcc need some small but necessary runtime libaray such as libtcc1.a
    [libtcc 0.2.0]  and some header files, which should be found in [prefix]/lib/tcc
    error: failed to run custom build command for `libtcc v0.2.0`

The trouble seems to be:
1. if cc finds the MSVC compiler, all the compiler arguments hardcoded in build.rs are wrong
2. the "[prefix]/lib/tcc" convention is for unix, not windows. So as a windows user, I have no idea where to put everything from tinycc so that the build script is able to pick it up

--- a/build.rs
+++ b/build.rs
@@ -54,8 +54,13 @@ fn main() {
         }
     }
 
-    println!("cargo:rustc-link-lib=static=tcc");
+    if target.contains("msvc") {
+        println!("cargo:rustc-link-lib=static=libtcc");
+    } else {
+        println!("cargo:rustc-link-lib=static=tcc");
+    }
     println!("cargo:rerun-if-changed=build.rs");
+    println!("cargo:rerun-if-env-changed=LIB_TCC");
 }
 
 fn build_tcc(config_arg: Option<&[&str]>, make_arg: Option<&[&str]>) {

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -531,6 +531,7 @@ mod tests {
     }
 
     #[test]
+    #[cfg(not(target_env = "msvc"))]
     fn link_lib() {
         let p = CString::new(
             r#"

